### PR TITLE
Fix measuring request duration

### PR DESCRIPTION
**Pull request by hennevogel**
The `state` getter if overloaded and transforms the string to a symbol
(why???), we cover for that now.

Also simplifying the method a bit.

### CODE REVIEW

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -813,13 +813,14 @@ def setincident(incident)
...
+    return unless state_changed?
     # new->review && review->new are not worth an event - it's just spam
-    return if state.to_s.in?(intermediate_state) && state_was.to_s.in?(intermediate_state)
+    return unless conclusive?
 
     options = event_parameters
-    options[:duration] = (updated_at - created_at).to_i if conclusive?
+
+    # measure duration unless superseding a final state, like revoked -> superseded
+    options[:duration] = (updated_at - created_at).to_i if FINAL_REQUEST_STATES.exclude?(state_was.to_sym) && FINAL_REQUEST_STATES.exclude?(state)
```
Comment (krauselukas) [inline src/api/app/models/bs_request.rb:823]: @hennevogel the previous behaviour was `FINAL_REQUEST_STATES.exclude?(state_was.to_sym) && FINAL_REQUEST_STATES.include?(state)` is this change intended

Comment (krauselukas) [reply src/api/app/models/bs_request.rb:823]: so a state change from `:new -> :accepted` would not record the duration

Comment (hennevogel) [reply src/api/app/models/bs_request.rb:823]: Of course not. copy pasta :) Should be `exclude(was) && include(is)`

- `FINAL_REQUEST_STATES.exclude?(:new) => true`
- `FINAL_REQUEST_STATES.include?(:accepted) => true`

or 

- `FINAL_REQUEST_STATES.exclude?(:accepted) => false`
- `FINAL_REQUEST_STATES.include?(:superseded) => true`

Good catch!

**Review by krauselukas — APPROVED** [review]
